postgres-types/src/cidr_02.rs

```diff
@@ -0,0 +1,46 @@
+use bytes::BytesMut;
+use cidr_02::{IpCidr, IpInet};
+use postgres_protocol::types;
+use std::error::Error;
+
+use crate::{FromSql, IsNull, ToSql, Type};
+
+impl<'a> FromSql<'a> for IpCidr {
+    fn from_sql(_: &Type, raw: &[u8]) -> Result<Self, Box<dyn Error + Sync + Send>> {
+        let inet = types::inet_from_sql(raw)?;
+        Ok(IpCidr::new(inet.addr(), inet.netmask())
+            .expect("postgres cidr type has zeroed host portion"))
+    }
+
+    accepts!(CIDR);
+}
+
+impl ToSql for IpCidr {
+    fn to_sql(&self, _: &Type, w: &mut BytesMut) -> Result<IsNull, Box<dyn Error + Sync + Send>> {
+        types::inet_to_sql(self.first_address(), self.network_length(), w);
+        Ok(IsNull::No)
+    }
+
+    accepts!(CIDR);
+    to_sql_checked!();
+}
+
+impl<'a> FromSql<'a> for IpInet {
+    fn from_sql(_: &Type, raw: &[u8]) -> Result<Self, Box<dyn Error + Sync + Send>> {
+        let inet = types::inet_from_sql(raw)?;
+        Ok(IpInet::new(inet.addr(), inet.netmask())
+            .expect("postgres enforces maximum length of netmask"))
```

Comment:
Same

Reply:
Similar, postgres guarantees that the `inet` type has a mask length corresponding to the address type (<= 32 for IPv4, <=128 for IPv6)